docs/pages/config-plugins/mods.mdx

```diff
@@ -0,0 +1,362 @@
+
+#### File import
+
+You can quickly create a plugin in your project by creating a JavaScript/TypeScript and use it in your config like any other JS/TS file.
+
+<FileTree
+  files={[
+    ['app.config.ts', <code>import "./my-config-plugin"</code>],
+    [
+      'my-config-plugin.ts',
+      <>
+        <YesIcon /> Imported from config
+      </>,
+    ],
+  ]}
+/>
+
+In the above example, the config plugin file contains a bare minimum function:
+
+```ts my-config-plugin.ts
+module.exports = ({ config }: { config: ExpoConfig }) => {};
+```
+
+#### Inline function inside of dynamic app config
+
+Expo config objects also support passing functions as-is to the `plugins` array. This is useful for testing, or if you want to use a plugin without creating a file.
+
+```js app.config.ts
+const withCustom = (config, props) => config;
+
+const config = {
+  plugins: [
+    [
+      withCustom,
+      {
+        /* props */
+      },
+    ],
+    /* @info Without props */
+    withCustom,
+    /* @end */
+  ],
+};
+```
+
+One caveat to using functions instead of strings is that serialization will replace the function with the function's name. This keeps **manifests** (kind of like the **index.html** for your app) working as expected. Here is what the serialized config would look like:
+
+```json
+{
+  "plugins": [["withCustom", {}], "withCustom"]
+}
+```
+
+### Standalone package plugins
+
+> **info** See [Create a module with a config plugin](/config-plugin-and-native-module-tutorial/) for a step-by-step guide on how to create a standalone package plugin.
+
+Standalone package plugins can be implemented in two ways:
+
+#### 1. Dedicated config plugin packages
+
+These are npm packages whose sole purpose is to provide a config plugin. For a dedicated config plugin package, you can export your plugin using `app.plugin.js`:
+
+<FileTree
+  files={[
+    ['app.config.ts', <code>import "expo-splash-screen"</code>],
+    ['node_modules/expo-splash-screen', 'Node module'],
+    [
+      'node_modules/expo-splash-screen/app.plugin.js',
+      <>
+        <YesIcon /> Entry file for custom plugins
+      </>,
+    ],
+    [
+      'node_modules/expo-splash-screen/build/index.js',
+      <>
+        <NoIcon /> Skipped in favor of <code>app.plugin.js</code>
+      </>,
+    ],
+  ]}
+/>
+
+#### 2. Config plugins with companion packages
+
+When a config plugin is part of a Node module without an **app.plugin.js**, it uses the package's `main` entry point:
+
+<FileTree
+  files={[
+    ['app.config.ts', <code>import "expo-splash-screen"</code>],
+    ['node_modules/expo-splash-screen', 'Node module'],
+    ['node_modules/expo-splash-screen/package.json', <code>"main": "./build/index.js"</code>],
+    [
+      'node_modules/expo-splash-screen/build/index.js',
+      <>
+        <YesIcon /> Node resolve to this file
+      </>,
+    ],
+  ]}
+/>
+
+#### Plugin resolution order
```

Comment:
maybe this should be an H3 so it isn't at the same level as the "implemented two ways" section?